core/src/main/java/hudson/model/Computer.java

```diff
@@ -1467,6 +1467,11 @@ public void doConfigSubmit( StaplerRequest req, StaplerResponse rsp ) throws IOE
             throw new FormException(Messages.ComputerSet_SlaveAlreadyExists(proposedName), "name");
         }
 
+        String nExecutors = req.getSubmittedForm().getString("numExecutors");
+        if (!nExecutors.matches("^(\\d*[1-9]\\d*)$")) {
```

Comment:
`^` and `$` are gratuitous when using `matches` (as opposed to, say, `Matcher.find`). So are the `()`, since you are not asking for any `group`.

More to the point, this will improperly reject `0` AFAICT, which is a valid executor number (for the master node anyway). Or is this code called only for agents?

Maybe simpler to just `Integer.parseInt`, catch `NumberFormatException`, and reject a negative result.